source/1_engine/rex_engine/include/rex_engine/function_signature_define.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
```

Comment:
don't think this file should be in the root of rex_engine.
I assume this is copy pasted from spdlog? can it be cleaned up a bit?

we don't use compiler macros directly, we use REX_COMPILER_MSVC etc, please use those macros.
it's well defined which version of clang (15.0.0) and msvc (14.29.30133) we use.

I'd base the code around that instead of the below as it's not really clear what we're using

afaik __PRETTY_FUNCTION__  for anything but MSVC and __FUNCSIG__ for MSVC is fine nowadays